docs/SpellExample.java

```diff
@@ -0,0 +1,315 @@
+
+
+        /*
+         * Add some entries to the spells class
+         */
+
+        Value addFireResults = client.query(
+            Create(
+                Class(Value(SPELLS_CLASS)),
+                Obj("data",
+                    Obj("name", Value("Fire Beak"), "element", Value("water"), "cost", Value(15))
+                )
+            )
+        ).get();
+        System.out.println("Added spell to class " + SPELLS_CLASS + ":\n " + toPrettyJson(addFireResults) + "\n");
+
+
+        Value addDragonResults = client.query(
+            Create(
+                Class(Value(SPELLS_CLASS)),
+                Obj("data",
+                    Obj("name", Value("Water Dragon's Claw"), "element", Value("water"), "cost", Value(25))
+                )
+            )
+        ).get();
+        System.out.println("Added spell to class " + SPELLS_CLASS + ":\n " + toPrettyJson(addDragonResults) + "\n");
+
+        Value addHippoResults = client.query(
+            Create(
+                Class(Value(SPELLS_CLASS)),
+                Obj("data",
+                    Obj("name", Value("Hippo's Wallow"), "element", Value("water"), "cost", Value(35))
+                )
+            )
+        ).get();
+        System.out.println("Added spell to class " + SPELLS_CLASS + ":\n " + toPrettyJson(addHippoResults) + "\n");
+
+        Value.RefV hippoRef = addHippoResults.at("ref").to(Value.RefV.class).get();
+        System.out.println("hippoRef = " + hippoRef);
+
+        /*
+         * Read the hippo back that we just created
+         */
+
+        Value getHippoResults = client.query(
+            Get(Ref(
+                Class("spells"),
+                Value(hippoRef.getId())
+                )
+            )
+        ).get();
+        System.out.println("Hippo Spells:\n " + toPrettyJson(getHippoResults) + "\n");
+
+        //convert the hippo results into primitive elements
+        Value data = getHippoResults.get(Field.at("data"));
+        String element = data.get(Field.at("element")).to(String.class).get();
+        System.out.println("spell element = " + element);
+
+        Value.ObjectV objectV = getHippoResults.at("data").to(Value.ObjectV.class).get();
+        System.out.println("objectV = " + objectV);
+
+        /*
+         * Query for all the spells in the index
+         */
+        Value queryIndexResults = client.query(
+            Select(Value("data"),
+                Paginate(
+                    Match(Index(Value(INDEX_NAME)))
+                ))
+        ).get();
+        System.out.println("All spells:\n " + toPrettyJson(queryIndexResults) + "\n");
+
+        //convert the query results to a collection of refrences
+        Collection<Value.RefV> refVCollection = queryIndexResults.asCollectionOf(Value.RefV.class).get();
+        List<String> listOfIds = refVCollection.stream().map(refV -> refV.getId()).collect(Collectors.toList());
```

Comment:
instead of doing this map here you could do it on the server side using `SelectAll(Path("data", "id"), Paginate(...))`